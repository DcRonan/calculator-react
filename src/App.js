const App = () => {
  return (
    <div>
      <header>Calculator - React</header>
    </div>
  );
};

export default App;
