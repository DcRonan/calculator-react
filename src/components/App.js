/* eslint-disable no-unused-vars */
import React, { useState } from 'react';
import Navbar from './Navbar';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';
import Calculator from './Calculator';

const data = {
  total: '0',
  next: null,
  operation: null,
};

const App = () => {
  const [state, setState] = useState(data);

  const handleClick = buttonName => {
    const newState = calculate(state, buttonName);
    setState(newState);
  };

  const update = () => {
    const { total, next, operation } = state;

    if (next) return next;
    return total;
  };

  return (
    <>
      <header>
        <Navbar />
      </header>
      <main>
        <section className="calc-main">
          <div>
            <Calculator />
          </div>
          <div style={{ width: '70%' }}>
            <Display result={update()} />
            <ButtonPanel handleClick={handleClick} />
          </div>
        </section>
      </main>
    </>
  );
};

export default App;
