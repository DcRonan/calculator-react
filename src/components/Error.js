const Error = () => <h4>Error! Page not found.</h4>;

export default Error;
