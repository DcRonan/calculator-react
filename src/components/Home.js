import React from 'react';
import Navbar from './Navbar';

const Home = () => (
  <div>
    <Navbar />
    <h1>Hello from Home</h1>
  </div>
);

export default Home;
