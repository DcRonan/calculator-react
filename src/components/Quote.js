import Navbar from './Navbar';

const Quote = () => (
  <>
    <Navbar />
    <section className="quote">
      <h4 style={{ fontWeight: '500', fontSize: '30px' }}>
        Mathematics is not about numbers, equations, computations, or
        algorithms: it is about understanding. â€“William Paul Thurston
      </h4>
    </section>
  </>
);

export default Quote;
