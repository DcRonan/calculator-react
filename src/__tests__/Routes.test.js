import renderer from 'react-test-renderer';
import Route from '../components/Routes';
import Home from '../components/Home';
import App from '../components/App';
import Quote from '../components/Quote';
import Error from '../components/Error';

describe('checks if the Routes have the correct directories', () => {
  test('should render to Home route', () => {
    const component = Home;

    const tree = renderer
      .create(<Route path="/" component={component} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render to App route', () => {
    const component = App;

    const tree = renderer
      .create(<Route path="/calculator" component={component} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render to Quote route', () => {
    const component = Quote;

    const tree = renderer
      .create(<Route path="/calculator" component={component} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render to Error route', () => {
    const component = Error;

    const tree = renderer
      .create(<Route component={component} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
