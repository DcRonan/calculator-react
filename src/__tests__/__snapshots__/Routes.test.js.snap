// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if the Routes have the correct directories should render to App route 1`] = `
Array [
  <nav>
    <div>
      <div
        className="title"
      >
         Math Magicians 
      </div>
    </div>
    <div
      className="navbar"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </div>
  </nav>,
  <section
    style={
      Object {
        "padding": "50px 20px 30px 20px",
      }
    }
  >
    <h3
      style={
        Object {
          "padding": "0 0 30px 10px",
        }
      }
    >
      Welcome to our page!
    </h3>
    <div>
      <div
        style={
          Object {
            "paddingBottom": "20px",
          }
        }
      >
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum.
      </div>
      <div>
        IpsuLorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
      </div>
    </div>
  </section>,
]
`;

exports[`checks if the Routes have the correct directories should render to Error route 1`] = `
Array [
  <nav>
    <div>
      <div
        className="title"
      >
         Math Magicians 
      </div>
    </div>
    <div
      className="navbar"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </div>
  </nav>,
  <section
    style={
      Object {
        "padding": "50px 20px 30px 20px",
      }
    }
  >
    <h3
      style={
        Object {
          "padding": "0 0 30px 10px",
        }
      }
    >
      Welcome to our page!
    </h3>
    <div>
      <div
        style={
          Object {
            "paddingBottom": "20px",
          }
        }
      >
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum.
      </div>
      <div>
        IpsuLorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
      </div>
    </div>
  </section>,
]
`;

exports[`checks if the Routes have the correct directories should render to Home route 1`] = `
Array [
  <nav>
    <div>
      <div
        className="title"
      >
         Math Magicians 
      </div>
    </div>
    <div
      className="navbar"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </div>
  </nav>,
  <section
    style={
      Object {
        "padding": "50px 20px 30px 20px",
      }
    }
  >
    <h3
      style={
        Object {
          "padding": "0 0 30px 10px",
        }
      }
    >
      Welcome to our page!
    </h3>
    <div>
      <div
        style={
          Object {
            "paddingBottom": "20px",
          }
        }
      >
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum.
      </div>
      <div>
        IpsuLorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
      </div>
    </div>
  </section>,
]
`;

exports[`checks if the Routes have the correct directories should render to Quote route 1`] = `
Array [
  <nav>
    <div>
      <div
        className="title"
      >
         Math Magicians 
      </div>
    </div>
    <div
      className="navbar"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </div>
  </nav>,
  <section
    style={
      Object {
        "padding": "50px 20px 30px 20px",
      }
    }
  >
    <h3
      style={
        Object {
          "padding": "0 0 30px 10px",
        }
      }
    >
      Welcome to our page!
    </h3>
    <div>
      <div
        style={
          Object {
            "paddingBottom": "20px",
          }
        }
      >
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum.
      </div>
      <div>
        IpsuLorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
      </div>
    </div>
  </section>,
]
`;
